src/tests/conformance/post_handshake/broadcast/proposalpayload.rs

```diff
@@ -0,0 +1,44 @@
+use tempfile::TempDir;
+
+use crate::{
+    protocol::codecs::payload::Payload, setup::node::Node,
+    tools::synthetic_node::SyntheticNodeBuilder,
+};
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn c007_PROPOSAL_PAYLOAD_expect_after_connect() {
+    // ZG-CONFORMANCE-007
+
+    // Spin up a node instance.
+    let target = TempDir::new().expect("couldn't create a temporary directory");
+    let mut node = Node::builder()
+        .build(target.path())
+        .expect("unable to build the node");
+    node.start().await;
+
+    // Create a synthetic node and enable handshaking.
+    let mut synthetic_node = SyntheticNodeBuilder::default()
+        .build()
+        .await
+        .expect("unable to build a synthetic node");
+
+    let net_addr = node.net_addr().expect("network address not found");
+
+    // Connect to the node and initiate the handshake.
+    synthetic_node
+        .connect(net_addr)
+        .await
+        .expect("unable to connect");
+
+    let check = |m: &Payload| matches!(&m, Payload::ProposalPayload(..));
+
+    // Wait for two messages at least.
+    for _ in 0..2 {
```

Comment:
Are two `ProposalPayload` messages expected? If so why?

Reply:
> conformance/post_handshake/**broadcast**/proposalpayload.rs

Because this is a broadcast message, so maybe it's not necessary, but I thought this could be a funny way to indicate a frequent broadcast: by waiting for at least two messages (we receive both within two seconds)

I could wait for one message or leave a comment about why we wait for two, wdyt?

Reply:
Sure that seems fine to me. It's a fairly weak guarantee but it's worth sanity checking.